exonum-java-binding-core/rust/src/utils/jni_cache.rs

```diff
@@ -0,0 +1,158 @@
+        THROWABLE_CACHED_GET_MESSAGE = get_method_id(
+            &env,
+            "java/lang/Throwable",
+            "getMessage",
+            "()Ljava/lang/String;",
+        );
+        UTA_CACHED_EXECUTE = get_method_id(
+            &env,
+            "com/exonum/binding/service/adapters/UserTransactionAdapter",
+            "execute",
+            "(J)V",
+        );
+        UTA_CACHED_INFO = get_method_id(
+            &env,
+            "com/exonum/binding/service/adapters/UserTransactionAdapter",
+            "info",
+            "()Ljava/lang/String;",
+        );
+        UTA_CACHED_VERIFY = get_method_id(
+            &env,
+            "com/exonum/binding/service/adapters/UserTransactionAdapter",
+            "isValid",
+            "()Z",
+        );
+        USA_CACHED_STATE_HASHES = get_method_id(
+            &env,
+            "com/exonum/binding/service/adapters/UserServiceAdapter",
+            "getStateHashes",
+            "(J)[[B",
+        );
+        USA_CACHED_CONVERT_TRANSACTION = get_method_id(
+            &env,
+            "com/exonum/binding/service/adapters/UserServiceAdapter",
+            "convertTransaction",
+            "([B)Lcom/exonum/binding/service/adapters/UserTransactionAdapter;",
+        );
+        CLASS_CACHED_ERROR = env
+            .new_global_ref(env.find_class("java/lang/Error").unwrap().into())
+            .ok();
+        CLASS_CACHED_TEE = env
+            .new_global_ref(
+                env.find_class("com/exonum/binding/transaction/TransactionExecutionException")
+                    .unwrap()
+                    .into(),
+            ).ok();
+
+        assert!(
+            OBJECT_CACHED_GET_CLASS.is_some()
+                && CLASS_CACHED_ERROR.is_some()
+                && THROWABLE_CACHED_GET_MESSAGE.is_some()
+                && UTA_CACHED_EXECUTE.is_some()
+                && UTA_CACHED_INFO.is_some()
+                && UTA_CACHED_VERIFY.is_some()
+                && USA_CACHED_STATE_HASHES.is_some()
+                && USA_CACHED_CONVERT_TRANSACTION.is_some()
+                && CLASS_CACHED_ERROR.is_some()
+                && CLASS_CACHED_TEE.is_some(),
+            "Error caching Java entities"
+        );
+
+        info!("Done caching references to Java classes and methods.");
+    }
+}
+
+// Helper method. Produces `JMethodID` for a particular class dealing with lifetime.
+fn get_method_id(env: &JNIEnv, class: &str, name: &str, sig: &str) -> Option<JMethodID<'static>> {
+    env.get_method_id(class, name, sig)
+        // we need this line to erase lifetime in order to save underlying raw pointer in static
+        .map(|mid| mid.into_inner().into())
+        .ok()
+}
+
+/// Returns cached `JClass` for `java/lang/Error` as a `GlobalRef`
+pub fn get_error_class() -> GlobalRef {
+    unsafe { CLASS_CACHED_ERROR.clone().unwrap() }
+}
+
+/// Returns cached `JClass` for `TransactionExecutionException` as a `GlobalRef`
+pub fn get_tee_class() -> GlobalRef {
```

Comment:
I believe `get_` prefixes are redundant in case you always import `jni_cache`

Reply:
Thanks, I also feel like that, let's also wait for anyone else's opinion.